protoplugin.go

```diff
@@ -145,6 +145,15 @@ func WithLenientValidation(lenientValidateErrorFunc func(error)) RunOption {
 	})
 }
 
+// WithUnmarshalOptions returns a new RunOption that overrides the default proto.UnmarshalOptions for decoding
+// the CodeGeneratorRequest. Among other uses, this is useful if you need to control how extensions are resolved
+// in the CodeGeneratorRequest.
+func WithUnmarshalOptions(unmarshalOptions proto.UnmarshalOptions) RunOption {
```

Comment:
Instead of exposing all the unmarshal options, it seems like what you want to do is expose the ability to have a custom resolver

Reply:
Makes sense, I was on the fence about it since it is a little awkward to test. I pushed a commit that switches to only adding an extension type resolver option.